### PR TITLE
Added FinderSync Extension

### PR DESCRIPTION
**Pull request by Camji55**
Added FinderSync Extension. To use the app now, the user must open the app (This can be set to launch automatically on login) then add the tool by customizing the finder toolbar and adding "Open In Terminal". The Icon will swift with between dark mode (Icon needs to be updated).

Let me know if you have any questions.

### CODE REVIEW

**Comment by Camji55** [comment]
Finder hard reset is necessary to change the icon for now,

**Comment by Ji4n1ng** [comment]
You really did a great job! I learned a lot from your code. I have one question about how to get the FinderSync Extension icon to change. Does the "Finder hard reset" you said above mean "force the Finder to exit and restart"? Currently I can only change the icon by this way.

**Comment by Ji4n1ng** [comment]
I have a few personal opinions I want to share with you.

1. I think simplicity and convenience are more important for a small tool application. FinderSync Extension requires two clicks. Even if it's just two small clicks, I think it will increase the cost of user usage. Some people may not like it. They may only want stability and ease of use.
2. If Mojave doesn't provide more APIs to let us change the icons, I don't think we need to do this. After all, it's not a core feature. And WWDC is not far away. Perhaps there will be more convenient APIs for us to use on 10.15. At least I hope so.
3. In addition FinderSync is a great way to extend the functionality of the app. I think there may be two forms of apps available for users to choose from. One is that, as it is now, user can only open the path to the terminal. The other is in the form of FinderSync Extension, which is more powerful. In the menu of FinderSync, the user can choose not only to open the terminal, but also to copy the current path to the clipboard, and so on.

In short, the current version, which is a simple version, will continue to be maintained in the form of v0.2.0, if there is no elegant way to solve the dark icon problem. And FinderSync Extension will appear in the powerful version.

---

These are just my personal opinions. How the `OpenInTerminal` develops requires everyone's opinion. This is my first time to do an open source project seriously. I really want to know your point of view.

Finally, thank you very much for your contribution. You are really great.

Looking forward to your reply. 😃

**Comment by Camji55** [comment]
That's great that you learned a lot, swift development is my full-time job and part-time job 😉.

To answer your questions:
1. When I say force restart I do mean relaunching it from the Force Quit menu.
2. In terms of simplicity, I completely agree about the 2 clicks being an unnecessary extra step for a lot of users. Unfortunately I was unable to find an elegant solution to this problem (Seems I must have the drop down menu instead of just a single action when the button is clicked.)
3. In terms of the icons, I can play around with it a little more. The current way it is handled in current versions cannot currently be dynamically changed based on dark/light mode, however, the finder sync extension icon can (hence the trouble of creating the extension in the first place.)
4. In terms of 2 versions of the app, I kind of agree in the sense of letting the user choose how they wish to use the app, but I think they could be bundled into the same application. I'd have to look more into this but for example if they want simple functionality, they drag the app into the toolbar. If they want advanced functionality they use the FinderSync extension.

Thanks for your kind words! I really enjoy this app. There are a few others out there but none as well thought out as this one appears to have been. Nice work 👍🏻

**Comment by Camji55** [comment]
Added another small little feature to it. Check my last commit.
